### PR TITLE
Add devices endpoint

### PR DESCRIPTION
**Pull request by iinuwa**
Adds 5 device endpoints:
* get_device
* get_devices
* set_device
* delete_device
* bulk_delete_devices

Addresses two compatibility items:
* POST /delete_devices (#49)
* Add 'Device Management' module (#48)

### CODE REVIEW

**Comment by iinuwa** [comment]
I noticed that the `register` and `login` endpoints use the `String` type instead of using the `DeviceId` alias for the `device_id` parameter: should we prefer one over the other?

**Comment by jplatte** [comment]
We should be using `DeviceId`. Will leave more comments later when I find the time.

File: src/r0/device/bulk_delete_devices.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    request {
+        /// List of devices to delete.
+        devices: Vec<DeviceId>,
+
+        /// Additional authentication information for the user-interactive authentication API.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        auth: Option<AuthenticationData>,
+    }
+
+    response { }
```
Comment (jplatte) [inline src/r0/device/bulk_delete_devices.rs]: Just to have a consistent style:

```suggestion
    response {}
```

File: src/r0/device/bulk_delete_devices.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+use crate::r0::account::AuthenticationData;
+use ruma_api::ruma_api;
+use ruma_identifiers::DeviceId;
+
+ruma_api! {
+    metadata {
+        description: "Delete specified devices.",
+        method: POST,
+        path: "/_matrix/client/r0/delete_devices",
+        name: "bulk_delete_devices",
```
Comment (jplatte) [inline src/r0/device/bulk_delete_devices.rs]: I'd just name this `delete_devices`.

File: src/r0/device/get_devices.rs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        name: "get_devices",
+        path: "/_matrix/client/r0/devices",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    request {}
+
+    response {
+        devices: Vec<Device>
```
Comment (jplatte) [inline src/r0/device/get_devices.rs]:
```suggestion
        devices: Vec<Device>,
```

File: src/r0/device/set_device.rs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+//! [PUT /_matrix/client/r0/devices/{deviceId}](https://matrix.org/docs/spec/client_server/r0.6.0#put-matrix-client-r0-devices-deviceid)
+
+use ruma_api::ruma_api;
+use ruma_identifiers::DeviceId;
+
+ruma_api! {
+    metadata {
+        description: "Update metadata for a device.",
+        method: PUT,
+        name: "set_device",
```
Comment (jplatte) [inline src/r0/device/set_device.rs]: This should be named `update_device` or `update_device_metadata`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -19,3 +19,4 @@ Improvements:
 * Add `contains_url` to `r0::filter::RoomEventFilter` (introduced upstream in r0.3.0)
 * Update `r0::account::change_password` from r0.3.0 to r0.6.0
   * Add optional `auth` field
+* Add `r0::device` endpoint
```
Comment (jplatte) [inline CHANGELOG.md]:
```suggestion
* Add `r0::device` endpoints 
```

**Comment by iinuwa** [comment]
Thanks for the review! These should all be addressed now.

**Comment by jplatte** [comment]
Thanks!

**Pull request merged**